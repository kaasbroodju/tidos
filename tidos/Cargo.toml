[package]
name = "tidos"
version = { workspace = true }
authors = ["kaasbroodju"]
license = "MIT"
repository = "https://github.com/kaasbroodju/tidos"
description = "Tidos is a component framework that can be served by any web framework."
readme = "../README.md"
rust-version.workspace = true
edition.workspace = true
keywords = ["tidos", "web", "framework", "component"]
categories = ["web-programming"]

[package.metadata.docs.rs]
all-features = true

[dependencies]
rocket = { version = "0.5.1", optional = true }
tidos_macro = { workspace = true }

[dev-dependencies]
trybuild = "1.0.104"

[features]
rocket = ["dep:rocket"]
