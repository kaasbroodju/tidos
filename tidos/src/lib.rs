//! # Tidos - Documentation
//! Tidos is a high-performance Rust-based component framework that seamlessly integrates with any web framework, enabling developers to build dynamic web applications with ease. With Tidos’ powerful macros, you can intuitively create components directly within your Rust code. It even allows you to leverage Rust's pattern matching, loops, and conditionals inside your components—making your UI logic more expressive and maintainable.

mod components;
mod page;

#[doc(hidden)]
pub mod internals;

pub use components::Component;
#[doc(hidden)]
pub use internals::sanitize;
pub use page::Page;
pub use tidos_macro::*;

#[doc(hidden)]
#[macro_export]
macro_rules! sanitize {
	($input:expr) => {
		&tidos::internals::sanitize(String::from($input))
	};
}
