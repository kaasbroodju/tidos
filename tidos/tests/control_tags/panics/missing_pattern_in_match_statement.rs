use tidos::view;

fn a_complex_match_statement() {
	enum Pet {
		Fish,
		Dog,
		Cat,
		Other { name: String }
	}
	use Pet::*;
	let my_pet = Other { name: String::from("spider") };

	assert_eq!(
		&view! {
			{#match my_pet}
				{:case Fish}
					<p>{"Blub!"}</p>
				{:case Dog}
					<p>{"Good boy!"}</p>
				{:case Cat}
					<p>{"Give al mortal possessions to cat!"}</p>
					<p>{"Give al mortal possessions to cat!"}</p>
					<p>{"Give al mortal possessions to cat!"}</p>
			{/match}
		},
		"<p>What is that?</p><p>It is a spider!</p>"
	)
}

fn main() {}