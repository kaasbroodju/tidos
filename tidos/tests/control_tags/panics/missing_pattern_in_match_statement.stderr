error[E0004]: non-exhaustive patterns: `Pet::Other { .. }` not covered
  --> tests/control_tags/panics/missing_pattern_in_match_statement.rs:15:12
   |
15 |             {#match my_pet}
   |                     ^^^^^^ pattern `Pet::Other { .. }` not covered
   |
note: `Pet` defined here
  --> tests/control_tags/panics/missing_pattern_in_match_statement.rs:4:7
   |
4  |     enum Pet {
   |          ^^^
...
8  |         Other { name: String }
   |         ----- not covered
   = note: the matched value is of type `Pet`
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown
   |
25 ~         },
26 ~         Pet::Other { .. } => todo!(),
   |
